### PR TITLE
Cache values when accessing test dependencies from live context

### PR DESCRIPTION
**Pull request by oronbz**
#77

### CODE REVIEW

File: Tests/DependenciesTests/DependencyValuesTests.swift [hunk 1]
```diff
@@ -185,22 +185,17 @@ final class DependencyValuesTests: XCTestCase {
...
-                """
-                @Dependency(\\.reuseClient)" has no live implementation, but was accessed from a live \
-                context.
-                """
-              )
-            }
-          #endif
-          XCTAssertEqual(reuseClient.count(), 0)
-        }
+        #if DEBUG
```
Comment (oronbz) [inline Tests/DependenciesTests/DependencyValuesTests.swift:188]: The modifications here are to adjust to the fact the failure only happens once

**Comment by oronbz** [comment]
@stephencelis @mbrandonw , hope it's ok mentioning you here. any chance you can look into it? As our project switch more and more dependencies to swift-dependencies this become a bigger of an issue for us.

**Review by stephencelis — APPROVED** [review]
@oronbz Sorry for the delay. We've dug a bit more deeply into the problem and agree that we should be caching in this case. Thanks again for the PR!

**Comment by oronbz** [comment]
Lovely. Thanks @stephencelis

**Pull request merged**